src/components/SearchResultDetail/SearchResultDetail.tsx

```diff
@@ -4,16 +4,42 @@ import * as React from 'react';
 
 import ClipLoader from 'react-spinners/ClipLoader';
 
-import { findSearchResultAction } from '@/app/actions/action';
+import { findSearchResultAction, exportSearchResultPdfAction } from '@/app/actions/action';
 import { SearchResultStatus } from '@/constants/enum';
 import { useModal } from '@/hooks/useModal';
 import toast from '@/lib/toast';
 
 import SearchResultHtmlModal from './SearchResultHtmlModal';
 
-const SearchResultDetail = ({ id }: { id: string }) => {
+interface DownloadBase64Params {
+  base64String: string;
+  fileName: string;
+  mimeType: string;
+}
+
+const mimeTypePDF = 'application/pdf';
+
+const downloadBase64String = (downloadParams: DownloadBase64Params) => {
+  const { base64String, fileName, mimeType } = downloadParams;
+
+  const byteCharacters = Buffer.from(base64String, 'base64');
+  const downloadUrl = window.URL.createObjectURL(new Blob([byteCharacters], { type: mimeType }));
+  const link = document.createElement('a');
+  link.href = downloadUrl;
+  link.setAttribute('download', fileName);
+
+  link.click();
+};
+
+interface SearchResultDetailProps {
+  id: string;
+}
+
+const SearchResultDetail = (props: SearchResultDetailProps) => {
+  const { id } = props;
   const [searchResult, setSearchResult] = React.useState<any>(null);
   const [loading, setLoading] = React.useState<boolean>(false);
+  const [downloadLoading, setDownloadLoading] = React.useState<boolean>(false);
```

Comment:
`setDownloadLoading` is quite a moutful 😅 Hwo about:

```suggestion
  const [downloadInProgress, setDownloadInProgress] = React.useState<boolean>(false);
```